buildenv/jenkins/common/variables-functions.groovy

```diff
@@ -1436,4 +1433,31 @@ def create_job(JOB_NAME, SDK_VERSION, SPEC, downstreamJobType, id) {
         3, 120)
 }
 
+def set_build_variables_per_node() {
+    BOOT_JDK = check_path(buildspec.getScalarField('boot_jdk', SDK_VERSION), true)
+    println("BOOT_JDK: ${BOOT_JDK}")
+    if (!BOOT_JDK) {
+        error("BOOT_JDK: ${BOOT_JDK} does not exist!")
+    }
+
+    def freemarkerPath = buildspec.getScalarField('freemarker', SDK_VERSION)
+    FREEMARKER = check_path(buildspec.getScalarField('freemarker', SDK_VERSION), false)
+    println("FREEMARKER: ${FREEMARKER}")
+    if (!FREEMARKER) {
+        error("FREEMARKER: ${FREEMARKER} does not exist!")
+    }
+
+    OPENJDK_REFERENCE_REPO = check_path(buildspec.getScalarField("openjdk_reference_repo", SDK_VERSION), true)
+    println("OPENJDK_REFERENCE_REPO: ${OPENJDK_REFERENCE_REPO}")
```

Comment:
Do we want to have a slightly more detailed message here instead of just blank?  Maybe a 'not configured for this node' to be more meaningful if its not present?

Reply:
We could add a warning, but should stop the build.

Reply:
Agreed.  I like this change, 99% of the time nobody will notice, but for those few of us that debug to figure out why jobs (or clone) are slow this will be important information.